packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,363 @@
+import type { AnonCredsCredentialValue } from '@aries-framework/anoncreds'
+import type { Agent, FileSystem } from '@aries-framework/core'
+import type { EntryObject } from '@hyperledger/aries-askar-shared'
+
+import { AnonCredsCredentialRecord, AnonCredsLinkSecretRecord } from '@aries-framework/anoncreds'
+import { AskarWallet } from '@aries-framework/askar'
+import { InjectionSymbols, KeyDerivationMethod, JsonTransformer, TypedArrayEncoder } from '@aries-framework/core'
+import { Key, KeyAlgs, Store, StoreKeyMethod } from '@hyperledger/aries-askar-shared'
+
+import { IndySdkToAskarMigrationError } from './errors/IndySdkToAskarMigrationError'
+import { transformFromRecordTagValues } from './utils'
+
+/**
+ *
+ * Migration class to move a wallet form the indy-sdk structure to the new
+ * askar wallet structure.
+ *
+ * Right now, this is ONLY supported within React Native environments AND only sqlite.
+ *
+ * The reason it only works within React Native is that we ONLY update the
+ * keys, masterSecret and credentials for now. If you have an agent in Node.JS
+ * where it only contains these records, it may be used but we cannot
+ * guarantee a successful migration.
+ *
+ */
+export class IndySdkToAskarMigrationUpdater {
+  private store: Store
+  private walletId: string
+  private defaultLinkSecretId: string
+  private agent: Agent
+  private dbPath: string
+  private fs: FileSystem
+  private deleteOnFinish: boolean
+
+  private constructor(
+    store: Store,
+    walletId: string,
+    agent: Agent,
+    dbPath: string,
+    deleteOnFinish = false,
+    defaultLinkSecretId?: string
+  ) {
+    this.store = store
+    this.walletId = walletId
+    this.dbPath = dbPath
+    this.agent = agent
+    this.fs = this.agent.dependencyManager.resolve<FileSystem>(InjectionSymbols.FileSystem)
+    this.defaultLinkSecretId = defaultLinkSecretId ?? walletId
+    this.deleteOnFinish = deleteOnFinish
+  }
+
+  public static async initialize({
+    dbPath,
+    agent,
+    deleteOnFinish,
+    defaultLinkSecretId,
+  }: {
+    dbPath: string
+    agent: Agent
+    deleteOnFinish?: boolean
+    defaultLinkSecretId?: string
+  }) {
+    const {
+      config: { walletConfig },
+    } = agent
+    if (!walletConfig) throw new IndySdkToAskarMigrationError('Wallet config is required for updating the wallet')
```

Comment:
Can we add a check that the `walletConfig.storage` is undefined (meaning default sqlite is used) or `walletConfig.storage.type` is `sqlite` and if not throw an error? That way it will fail if you try to do this with a postgres config.

Reply:
Yes, good catch! I am all for more strictness on this one.